receipt_upload/receipt_upload/line_items/reconstructor.py

```
@@ -58,6 +59,178 @@ def _amount(t: str) -> float | None:
+            ):
+                candidates.append((k, lab, val))
+                break
+    if not candidates:
+        return [], []
+
+    # Clean line-total mass: existing LINE_TOTAL labels + what geometry recovers
+    # with the current (SUBTOTAL/TAX-excluding) band.
+    lt_keys = {
+        k
+        for k, labs in label_objs.items()
+        if any(l.label == "LINE_TOTAL" for l in labs)
```

Ruff (0.15.17)

[error] Ambiguous variable name: `l`

(E741)